tpm2/structures.go

```diff
@@ -125,14 +130,24 @@ func (p *RSAParams) encode() ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	if p.Modulus == nil {
-		return nil, errors.New("RSAParams.Modulus must be set")
+	rest, err := tpmutil.Pack(p.KeyBits, p.Exponent)
+	if err != nil {
+		return nil, err
 	}
-	rest, err := tpmutil.Pack(p.KeyBits, p.Exponent, p.Modulus.Bytes())
+
+	if p.Modulus == nil && len(p.ModulusRaw) == 0 {
+		return nil, errors.New("RSAParams.Modulus or RSAParams.ModulusRaw must be set")
```

Comment:
Is it also an error for both to be set? If so, then please check for this (instead of clobbering mod in the code below).

Reply:
Yep, it's an error.
Not sure how i can clean up mod below.